ocrd_models/ocrd_models/ocrd_file.py

```diff
@@ -32,15 +34,12 @@ def __init__(self, el, mimetype=None, instance=None, local_filename=None, mets=N
         self._instance = instance
         self.mets = mets
 
-        if self.url:
-            if not self.url.startswith('http://') and \
-                not self.url.startswith('https://'):
-                self.local_filename = self.url
-            if self.url.startswith('file://'):
-                self.local_filename = self.url[len('file://'):]
+        if url:
+            self.url = url
 
-        #  if baseurl and not local_filename and '://' not in self.url:
-        #      self.local_filename = '%s/%s' % (baseurl, self.url)
+        if not(local_filename):
+            if self.url and is_local_filename(self.url):
```

Comment:
Is that for cases where the user incorrectly uses the URL field for local filenames? If yes do we really have to rescue them?

Reply:
> incorrectly uses the URL field for local filenames

No, this is for `mets:file` with `mets:Locat` with `@xlink:href` pointing to relative files.